src/main/kotlin/dev/isxander/modstitch/util/AccessWidener.kt

```diff
@@ -0,0 +1,492 @@
+                val className = parts[1].replace('.', '/')
+                val descStart = if (parts.size == 3) parts[2].indexOf('(') else -1
+                val (entryType, name, descriptor) = when {
+                    parts.size == 4 -> Triple(if (parts[3].startsWith('(')) AccessWidenerEntryType.METHOD else AccessWidenerEntryType.FIELD, parts[2], parts[3])
+                    parts.size == 3 && descStart < 0 -> Triple(AccessWidenerEntryType.FIELD, parts[2], "")
+                    parts.size == 3 -> Triple(AccessWidenerEntryType.METHOD, parts[2].substring(0, descStart), parts[2].substring(descStart))
+                    else -> Triple(AccessWidenerEntryType.CLASS, className, className)
+                }
+
+                entries.add(AccessWidenerEntry(entryType, AccessModifier(accessType, isFinal = isFinal), className, name, descriptor))
+                line = reader.readUncommentedLine()
+            }
+
+            return AccessWidener(AccessWidenerFormat.AT, entries)
+        }
+
+        /**
+         * Merges and deduplicates provided entries.
+         *
+         * @param entries The list of access widener entries to compact.
+         * @return A compacted list of deduplicated entries.
+         */
+        private fun compactEntries(entries: List<AccessWidenerEntry>): List<AccessWidenerEntry> {
+            val entryMap = mutableMapOf<AccessWidenerEntryIdentity, AccessModifier>()
+            for (entry in entries) {
+                val identity = AccessWidenerEntryIdentity(entry.type, entry.className, entry.name, entry.descriptor)
+                val curMod = entry.accessModifier
+                val oldMod = entryMap[identity] ?: curMod
+                val newMod = AccessModifier(
+                    if (curMod.type > oldMod.type) curMod.type else oldMod.type,
+                    curMod.isTransitive || oldMod.isTransitive,
+                    when {
+                        curMod.isFinal == false || oldMod.isFinal == false -> false
+                        curMod.isFinal == true || oldMod.isFinal == true -> true
+                        else -> null
+                    },
+                )
+                entryMap[identity] = newMod
+            }
+            return entryMap.map { AccessWidenerEntry(it.key.type, it.value, it.key.className, it.key.name, it.key.descriptor) }
+        }
+    }
+}
+
+/**
+ * Represents the supported access widener formats.
+ */
+internal enum class AccessWidenerFormat {
+    /** (Neo)Forge's `accesstransformer.cfg`. */
+    AT,
+
+    /** Fabric's `accessWidener v1`. */
+    AW_V1,
+
+    /** Fabric's `accessWidener v2`. */
+    AW_V2,
+}
+
+/**
+ * Defines the types of access modifiers that can be applied to classes, methods, or fields.
+ */
+internal enum class AccessModifierType {
+    /** The target keeps its original visibility modifier. */
+    NONE,
+
+    /** The target is made private. */
+    PRIVATE,
+
+    /** The target is made package-private. */
+    DEFAULT,
```

Comment:
can we just call this `PACKAGE_PRIVATE`? Default is ambiguous without looking at kdoc.
```suggestion
    PACKAGE_PRIVATE,
```